### PR TITLE
UCT: Cleanup interface operations definitions.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1865/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3832/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls review

**Comment by yosefe** [comment]
@MattBBaker it should work on ugni as well, but i would appreciate if you could test it too

**Comment by MattBBaker** [comment]
Build failure in UGNI, I'll track it down.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1884/ for details.

File: src/uct/ugni/rdma/ugni_rdma_iface.c [hunk 1]
```diff
@@ -161,54 +161,59 @@ static UCS_CLASS_CLEANUP_FUNC(uct_ugni_rdma_iface_t)
...
-
-static uct_iface_ops_t uct_ugni_gemini_rdma_iface_ops = {
-    .iface_query         = uct_ugni_rdma_iface_query,
+    .ep_fence            = uct_base_ep_fence,
+    .ep_create_connected = UCS_CLASS_NEW_FUNC_NAME(uct_ugni_ep_t),
+    .ep_destroy          = UCS_CLASS_DELETE_FUNC_NAME(uct_ugni_ep_t),
     .iface_flush         = uct_ugni_iface_flush,
+    .iface_fence         = uct_base_iface_fence,
+    .iface_progress_enable = ucs_empty_function,
+    .iface_progress_disable = ucs_empty_function,
```
Comment (MattBBaker) [inline src/uct/ugni/rdma/ugni_rdma_iface.c]: These two lines are the problem. It looks like there is a definition in uct.h for uct_iface_progress_enable() but uct_iface_ops_t does not have a .iface_progress_enable or a .iface_progress_disable. Removing these fixes everything and the tests pass.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3854/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1887/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1888/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3857/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@MattBBaker fixed

**Comment by yosefe** [comment]
@shamisp can u pls review?

**Comment by shamisp** [comment]
@MattBBaker please confirm

**Comment by shamisp** [comment]
@bbenton please take a look. The patch includes some API cleanup (mostly restructuring)

**Comment by shamisp** [comment]
Looks good to me.

**Comment by MattBBaker** [comment]
:+1: compiled and runs fine.

**Comment by bbenton** [comment]
:+1: for Doc/API

**Pull request merged**